app/containers/filterContainer/yearRangeSlider/index.tsx

```diff
@@ -0,0 +1,222 @@
+import { MIN_YEAR } from "./constants";
+import Slider from "./slider";
+import { SearchPageQueryParams } from "../../../components/articleSearch/types";
+import getQueryParamsObject from "../../../helpers/getQueryParamsObject";
+import PapersQueryFormatter from "../../../helpers/papersQueryFormatter";
+import { goToYearFilteredSearchResultPage, GoToYearFilteredSearchResultPageParams } from "./helper";
+import FilterButton from "../filterButton";
+const styles = require("./yearRangeSlider.scss");
+
+interface YearSet {
+  year: number;
+  docCount: number;
+}
+
+interface YearRangeSliderProps extends RouteComponentProps<null> {
+  yearInfo: YearSet[];
+  filteredYearInfo: YearSet[];
+}
+
+const Column: React.FunctionComponent<{
+  width: string;
+  height: string;
+  active: boolean;
+  yearSet: YearSet;
+  isSelecting: boolean;
+  isSelectMode: boolean;
+  filteredYearSet: YearSet | undefined;
+  filterHeight: string;
+  destinationParams: GoToYearFilteredSearchResultPageParams;
+}> = React.memo(props => {
+  return (
+    <>
+      {props.filteredYearSet &&
+        props.filteredYearSet.docCount > 0 && (
+          <div
+            style={{
+              width: props.width,
+              height: props.filterHeight,
+              backgroundColor: "#3e7fff",
+              borderRight: "solid 1px #f9f9fa",
+            }}
+            className={styles.filterColumn}
+            onClick={() => {
+              goToYearFilteredSearchResultPage(props.destinationParams);
+            }}
+          />
+        )}
+      <div
+        style={{
+          width: props.width,
+          height: props.height,
+          borderRight: "solid 1px #f9f9fa",
+        }}
+        className={classNames({
+          [styles.column]: true,
+          [styles.activeColumn]: props.active,
+        })}
+        onClick={() => {
+          goToYearFilteredSearchResultPage(props.destinationParams);
+        }}
+      >
+        <div
+          className={classNames({
+            [styles.columnLabel]: true,
+            [styles.isSelectMode]: props.isSelectMode,
+            [styles.isSelecting]: props.isSelecting,
+          })}
+        >
+          <div className={styles.docCount}>{`${props.yearSet.docCount} Papers`}</div>
+          <div className={styles.yearNum}>{`${props.yearSet.year} Year`}</div>
+        </div>
+      </div>
+    </>
+  );
+});
+
+const YearRangeSlider: React.FunctionComponent<YearRangeSliderProps> = props => {
+  const currentYear = new Date().getFullYear();
+  const yearSetListToShow: YearSet[] = React.useMemo(
+    () =>
+      new Array(currentYear - MIN_YEAR).fill("").map((_, i) => {
+        const yearSet = props.yearInfo.find(ys => ys.year === MIN_YEAR + i);
+        return {
+          year: MIN_YEAR + i,
+          docCount: yearSet ? yearSet.docCount : 0,
+        };
+      }),
+    [props.yearInfo]
+  );
+
+  const qp: SearchPageQueryParams = getQueryParamsObject(props.location.search);
```

Comment:
`props.location.search`

^ This variable seems to be used frequently. Why not declare it with a simpler variable name?

Reply:
or `const {location, history} = props;`or `React.FunctionComponent<YearRangeSliderProps> = {location, history ...} =>`

Reply:
Resolve.